sort/cocktailsort.go

```diff
@@ -0,0 +1,41 @@
+// Implementation of Cocktail sorting
+// reference: https://en.wikipedia.org/wiki/Cocktail_shaker_sort
+
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+// Cocktail sort is a variation of bubble sort, operating in two directions (beginning to end, end to beginning)
+func Cocktail[T constraints.Integer](arr []T) []T {
+	if len(arr) == 0 { // ignore 0 length arrays
+		return arr
+	}
+
+	swapped := false // true if swapped two or more elements in the last loop
+	// if it loops through the array without swapping, the array is sorted
+
+	for ok := true; ok; ok = swapped { // exits when swapped == false
+
+		for i := 0; i < len(arr)-1; i++ { // first loop, from array's beginning to end
```

Comment:
You probably need to set swapped to false before this loop.

Reply:
I thought about doing that but preferred the way I did instead. Maybe a more readable solution is to just put in a `for {}` with another if exit at the end.

Reply:
This code:
https://github.com/TheAlgorithms/Go/blob/4c573a955c61932ec1513d90f92f5177636fa3cb/sort/cocktailsort.go#L52
Is equivalent to:
```go
for swapped {
```
It just uses more code and locs, so I am not sure why you think its better.

Reply:
Done.